cmd/krel/cmd/anago/push_git_objects.go

```diff
@@ -0,0 +1,197 @@
+limitations under the License.
+*/
+
+package anago
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/release"
+	"k8s.io/release/pkg/util"
+)
+
+// pushGitObjectsCmd is the krel push-git-objects subcommand
+var pushGitObjectsCmd = &cobra.Command{
+	Use:   "push-git-objects",
+	Short: "Push branches and tags to the remote github repository",
+	Long: `krel push-git-objects
+
+NOTE: This subcommand is a temporary commando to be invoked from anago.
+
+The purpose of krel push-git-objects is to push branches and tags to the
+git remote repository of kubernetes. 
+`,
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := runPushGitObjects(pushGitObjectsOpts)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+type pushGitObjectsOptions struct {
+	nomock        bool
+	sayYes        bool
+	maxRetries    int
+	releaseBranch string
+	parentBranch  string
+	repoPath      string
+	tags          []string
+}
+
+var pushGitObjectsOpts = &pushGitObjectsOptions{}
+
+func init() {
+	pushGitObjectsCmd.PersistentFlags().StringVar(
+		&pushGitObjectsOpts.releaseBranch,
+		"release-branch",
+		"",
+		"branch name to push",
+	)
+
+	pushGitObjectsCmd.PersistentFlags().StringVar(
+		&pushGitObjectsOpts.parentBranch,
+		"parent-branch",
+		"",
+		fmt.Sprintf("parent branch, if different from %s", git.DefaultBranch),
+	)
+
+	pushGitObjectsCmd.PersistentFlags().BoolVarP(
+		&pushGitObjectsOpts.sayYes,
+		"yes",
+		"y",
+		false,
+		"say yes to prompts without asking the user",
+	)
+
+	pushGitObjectsCmd.PersistentFlags().BoolVar(
+		&pushGitObjectsOpts.nomock,
+		"nomock",
+		false,
+		"nomock flag",
+	)
+
+	pushGitObjectsCmd.PersistentFlags().StringSliceVarP(
+		&pushGitObjectsOpts.tags,
+		"tags",
+		"t",
+		[]string{},
+		"list of tags to push",
+	)
+
+	pushGitObjectsCmd.PersistentFlags().StringVar(
+		&pushGitObjectsOpts.repoPath,
+		"repo",
+		"/root/k8s",
+		"the local path to the repository to be used",
+	)
+
+	pushGitObjectsCmd.PersistentFlags().IntVar(
+		&pushGitObjectsOpts.maxRetries,
+		"max-retries",
+		10,
+		"number of times to retry git operations if a recoverable error occurs",
+	)
+
+	AnagoCmd.AddCommand(pushGitObjectsCmd)
+}
+
+func runPushGitObjects(options *pushGitObjectsOptions) (err error) {
+	if err := options.Validate(); err != nil {
+		return errors.Wrap(err, "validating command line options")
+	}
+	// Create the git pusher object
+	gitPusher, err := release.NewGitPusher(&release.GitObjectPusherOptions{
+		DryRun:     !options.nomock,
+		MaxRetries: options.maxRetries,
+		RepoPath:   options.repoPath,
+	})
+	if err != nil {
+		return errors.Wrap(err, "creating new git pusher object")
+	}
+
+	// # The real deal?
+	// Note: this --dry-run label is only for human readability
+	dryRunFlag := " --dry-run"
+	if options.nomock {
+		dryRunFlag = ""
+	}
```

Comment:
Yes I don't hink we need that. We can just print the `nomock` value in the logs / messages.

Reply:
OK, I've modified it to a human label of mock/nomock inside the git ops, it still says --dry-run (that's how the git package manages it)